sklearn/base.py

```diff
@@ -153,7 +154,20 @@ def _clone_parametrized(estimator, *, safe=True):
     return new_object
 
 
-class BaseEstimator(_HTMLDocumentationLinkMixin, _MetadataRequester):
+class _SampleWeightEstimatorMixin(_MetadataRequester):
+    """A Mixin to request ``sample_weight`` by default.
+
+    This Mixin makes the object to request ``sample_weight`` by default as ``True``
+    for both ``fit`` and ``score`` methods.
+
+    .. versionadded:: 1.7
+    """
+
+    __metadata_request__fit: dict[str, Union[bool, str]] = {"sample_weight": True}
+    __metadata_request__score: dict[str, Union[bool, str]] = {"sample_weight": True}
+
+
+class BaseEstimator(_HTMLDocumentationLinkMixin, _SampleWeightEstimatorMixin):
```

Comment:
I think this should apply to the estimators individually, and not `BaseEstimator`. This here would have a lot of impact on external libraries.